libs/cart/testing/src/in-memory-backend/cart.service.unit.spec.ts

```diff
@@ -0,0 +1,246 @@
+import { DaffProductImageFactory } from '@daffodil/product/testing';
+import { DaffCart } from '@daffodil/cart';
+
+import { DaffInMemoryBackendCartService } from './cart.service';
+import { DaffCartFactory } from '../../../testing/src';
+
+describe('DaffInMemoryBackendCartService | Unit', () => {
+  let cartTestingService: DaffInMemoryBackendCartService;
+  let stubCart: DaffCart;
+  let cartId;
+  let stubProductImage: DaffProductImage;
+  let daffCartFactory: jasmine.SpyObj<DaffCartFactory>;
+  let daffProductImageFactory: jasmine.SpyObj<DaffProductImageFactory>;
+
+  beforeEach(() => {
+    const daffCartFactorySpy = jasmine.createSpyObj('DaffCartFactory', ['create']);
+    const daffProductImageFactorySpy = jasmine.createSpyObj('DaffProductImageFactory', ['create']);
+    stubProductImage = new DaffProductImageFactory().create();
+    stubCart = new DaffCartFactory().create({image: stubProductImage});
+    cartId = stubCart.id;
+
+    TestBed.configureTestingModule({
+      providers: [
+        DaffInMemoryBackendCartService,
+        { provide: DaffCartFactory, useValue: daffCartFactorySpy},
+        { provide: DaffProductImageFactory, useValue: daffProductImageFactorySpy}
+      ]
+    });
+
+    daffCartFactory = TestBed.get(DaffCartFactory);
+    daffProductImageFactory = TestBed.get(DaffProductImageFactory);
+    daffProductImageFactory.create.and.returnValue(stubProductImage);
+    daffCartFactory.create.and.returnValue(stubCart);
+
+    cartTestingService = TestBed.get(DaffInMemoryBackendCartService);
+  });
+
+  it('should be created', () => {
+    expect(cartTestingService).toBeTruthy();
+  });
+
+  describe('after initialization', () => {
+    let result;
+
+    beforeEach(() => {
+      result = cartTestingService.createDb();
+    });
+
+    it('should have an empty array in DB', () => {
+      result = cartTestingService.createDb();
+      expect(result.cart).toEqual([]);
+    });
+  });
+
+  describe('processing a create request', () => {
+    let reqInfoStub;
+    let result;
+
+    beforeEach(() => {
+      reqInfoStub = {
+        id: 'create',
+        req: {
+          body: {}
+        },
+        utils: {
+          createResponse$: func => {
+            return func();
+          }
+        }
+      };
+
+      cartTestingService.carts.push(stubCart);
+    });
+
+    it('should return a partial with id', () => {
+      result = cartTestingService.post(reqInfoStub);
+
+      const id = result.body.id;
+
+      expect(id).toBeDefined();
+    });
+
+    it('should store a cart in the DB', () => {
+      result = cartTestingService.post(reqInfoStub);
+
+      const id = result.body.id;
+
+      expect(cartTestingService.carts.find(cart => cart.id === id)).toBeTruthy()
+    });
+  });
+
+  describe('processing a clear request', () => {
+    let reqInfoStub;
+    let result;
+
+    beforeEach(() => {
+      cartTestingService.carts.push(stubCart);
+      //add cartItems
+      cartTestingService.post({
```

Comment:
Can you just stub a cart with items in it instead?

Reply:
Changed in recent push.